contribs/drt/src/main/java/org/matsim/contrib/drt/prebooking/PrebookingStopActivity.java

```diff
@@ -84,27 +86,30 @@ protected void beforeFirstStep(double now) {
 	private void initDropoffRequests(double now) {
 		for (var request : dropoffRequests.values()) {
 			double leaveTime = now + stopDurationProvider.calcDropoffDuration(vehicle, request.getRequest());
-			leaveTimes.put(request.getId(), leaveTime);
+			leaveTimes.add(new RequestIdTime(request.getId(), leaveTime));
 		}
 
 		updateDropoffRequests(now);
 	}
 
 	private boolean updateDropoffRequests(double now) {
-		var iterator = leaveTimes.entrySet().iterator();
 
-		while (iterator.hasNext()) {
-			var entry = iterator.next();
-
-			if (entry.getValue() <= now) { // Request should leave now
-				passengerHandler.dropOffPassengers(driver, entry.getKey(), now);
-				prebookingManager.notifyDropoff(entry.getKey());
-				onboard -= dropoffRequests.get(entry.getKey()).getPassengerCount();
-				iterator.remove();
-			}
+		while (!leaveTimes.isEmpty() && leaveTimes.peek().time <= now) {
+			RequestIdTime rit = leaveTimes.poll();
```

Comment:
Actually, you don't even need `rit`, you can just save the `id` as `leaveTimes.poll().id`